### PR TITLE
Fix warnings in substrate

### PR DESCRIPTION
**Pull request by bkchr**

### CODE REVIEW

File: core/sr-io/without_std.rs [hunk 1]
```diff
@@ -59,16 +59,16 @@ pub extern fn oom(_: ::core::alloc::Layout) -> ! {
 /// See also "ext_" functions in sr-sandbox and sr-std
 extern "C" {
 	/// Most of the functions below return fixed-size arrays (e.g. hashes) by writing them into
-	/// memory regions that should be preallocated by module. 
+	/// memory regions that should be preallocated by module.
 	/// Functions that return variable-sized data use host-side allocations. These should be
 	/// manually freed by the module.
-	fn ext_free(addr: *mut u8);
-	
+	pub fn ext_free(addr: *mut u8);
```
Comment (pepyakin) [inline core/sr-io/without_std.rs]: I think we need to delete this. allocator crate uses its own externs

**Comment by pepyakin** [comment]
I'd like propose a TODO to add `RUSTFLAGS="-D warnings"` in CI config!

File: core/sr-io/without_std.rs (hunk 1 above)
Comment (bkchr) [reply core/sr-io/without_std.rs]: Okay

**Comment by bkchr** [comment]
@pepyakin ticket! :)

**Comment by bkchr** [comment]
@pepyakin you maybe want to review and merge?

File: core/cli/src/error.rs [hunk 2]
```diff
@@ -16,6 +16,9 @@
 
 //! Initialization errors.
 
+// Silence: `use of deprecated item 'std::error::Error::cause': replaced by Error::source, which can support downcasting`
+#![allow(deprecated)]
```
Comment (gnunicorn) [inline core/cli/src/error.rs]: if we use something deprecated don't we want to file a ticket for that and link it here and the other instances of this?

**Pull request merged**